arangod/Cluster/TakeoverShardLeadership.cpp

```diff
@@ -178,11 +179,12 @@ static void handleLeadership(LogicalCollection& collection, std::string const& l
       << "handling leadership of shard '" << databaseName << "/"
       << collection.name() << ": becoming leader";
 
-
-      // This will block the thread until we fetched a new current version
-      // in maintenance main thread.
-      feature.waitForLargerCurrentCounter(oldCounter);
       auto& ci = collection.vocbase().server().getFeature<ClusterFeature>().clusterInfo();
+      // This will block the thread until our ClusterInfo cache fetched a
+      // Current version in background thread which is at least as new as the
+      // Plan which brought us here. This is important for the assertion
+      // below where we check that we are in the list of failoverCandidates!
+      ci.waitForCurrent(planIndex);
```

Comment:
I think its true, just to be sure:
`waitForCurrent` also waits until the current index is really larger than `planIndex`, not just larger or equal to?

Reply:
No, it waits until the Version of `Current` in `ClusterInfo` is at least as new as the Raft index, which was underlying the Plan which triggered the `TakeOverShardLeadership` job.
But the Plan change in the agency will only ever be done **after** the new leader has been put into `failoverCandidates` in `Current`. Therefore, it will be there once `Current` has reached the Raft index of the Plan change.